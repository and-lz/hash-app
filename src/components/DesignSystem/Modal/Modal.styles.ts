import styled, { keyframes } from 'styled-components'

interface ModalProps {
  visible: boolean
}

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`
export const ModalBackground = styled.div<ModalProps>`
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: 999;
  background: rgba(0, 0, 0, 0.3);
  display: flex;
  justify-content: center;
  padding: 60px 100px 0;
  backdrop-filter: blur(5px);
  align-items: center;
  animation: ${fadeIn} 250ms ease 1 forwards;
`

const slide = keyframes`
  from {
    transform: translateY(100px);
    opacity: 0;
  }
  to {
    transform: translateY(0px);
    opacity: 1;
  }
`
export const Modal = styled.div`
  background-color: white;
  width: 380px;
  padding: 40px 50px;
  border-radius: 4px;
  overflow: auto;
  text-align: center;
  animation: ${slide} 250ms ease 100ms 1 both;
`
