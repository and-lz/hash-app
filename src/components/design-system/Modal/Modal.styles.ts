import styled from 'styled-components'

interface ModalProps {
  visible: boolean
}
export const ModalBackground = styled.div<ModalProps>`
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: 999;
  background: rgba(0, 0, 0, 0.2);
  display: flex;
  justify-content: center;
  padding: 60px 100px 0;
  backdrop-filter: blur(10px);
  align-items: center;
  pointer-events: none;
  opacity: 0;
  transition: all 250ms ease;
  ${props => {
    if (props.visible) {
      return `
        pointer-events: auto;
        opacity: 1;
      `
    }
  }}
`

export const Modal = styled.div`
  background-color: white;
  width: 400px;
  padding: 40px 50px;
  border-radius: 4px;
  overflow: auto;
  text-align: center;
`
