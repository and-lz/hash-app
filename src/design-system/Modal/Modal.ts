import styled, { keyframes } from 'styled-components'

interface ModalProps {
  visible: boolean
}

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
`
export const ModalBackground = styled.div<ModalProps>`
  position: absolute;
  width: 100%;
  height: 100%;
  z-index: 999;
  background: rgba(0, 0, 0, 0.1);
  display: flex;
  justify-content: center;
  padding: 20px;
  backdrop-filter: blur(5px);
  align-items: flex-end;
  animation: ${fadeIn} 250ms ease 1 forwards;
`

const slide = keyframes`
  from {
    transform: translateY(100px);
    opacity: 0;
  }
  to {
    transform: translateY(0px);
    opacity: 1;
  }
`
export const ModalBox = styled.div`
  background-color: white;
  width: 380px;
  border-radius: 4px;
  overflow: auto;
  text-align: center;
  border: 1px solid #d1dce3;
  animation: ${slide} 250ms ease 100ms 1 both;
`

export const ModalContent = styled.div`
  padding: 20px 50px;
`

export const ModalImage = styled.img.attrs({
  src: 'images/header.jpg',
})`
  width: 100%;
  margin-bottom: 0px;
  display: block;
`
