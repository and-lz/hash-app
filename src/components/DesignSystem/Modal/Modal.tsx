import React, { useEffect, useState } from 'react'
import Button from '../Button/Button'
import { Spacer } from '../Spaces/Spaces'
import { Heading1 } from '../Typography/Headings'
import { Text } from '../Typography/Paragraph'
import { ModalBackground, Modal as ModalBox } from './Modal.styles'

interface ModalProps {
  title: string
  content?: string
  visible: boolean
  onClose: Function
  action?: string
}

function Modal({
  title,
  content,
  visible,
  onClose,
  action = 'Fechar',
}: ModalProps) {
  const [showModal, setShowModal] = useState(visible)

  useEffect(() => {
    setShowModal(visible)
  }, [visible])

  if (!showModal) return <></>
  return (
    <>
      <ModalBackground visible={showModal}>
        <ModalBox>
          <Heading1>{title}</Heading1>
          <Text>{content}</Text>
          <Spacer size="big" />
          <Button
            label={action}
            onClick={() => {
              setShowModal(false)
              onClose()
            }}
          />
        </ModalBox>
      </ModalBackground>
    </>
  )
}

export default Modal
