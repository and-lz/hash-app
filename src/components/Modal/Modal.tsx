import Button from 'components/Button/Button'
import { Spacer } from 'design-system/Spaces/Spaces'
import { Heading1 } from 'design-system/Typography/Headings'
import { Text } from 'design-system/Typography/Paragraph'
import React, { useEffect, useState } from 'react'
import { Modal as ModalBox, ModalBackground } from 'design-system/Modal/Modal'

interface ModalProps {
  title: string
  content?: string
  visible?: boolean
  onClose?: Function
  action?: string
}

function Modal({
  title,
  content,
  visible = false,
  onClose = () => {},
  action = 'Fechar',
}: ModalProps) {
  const [showModal, setShowModal] = useState(visible)

  useEffect(() => {
    setShowModal(visible)
  }, [visible])

  if (!showModal) return <></>
  return (
    <>
      <ModalBackground visible={showModal}>
        <ModalBox>
          <Heading1 data-testid="modal-title">{title}</Heading1>
          <Text data-testid="modal-content">{content}</Text>
          <Spacer size="big" />
          <Button
            label={action}
            onClick={() => {
              setShowModal(false)
              onClose()
            }}
          />
        </ModalBox>
      </ModalBackground>
    </>
  )
}

export default Modal
