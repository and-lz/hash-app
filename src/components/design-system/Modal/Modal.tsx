import React, { useEffect, useState } from 'react'
import Button from '../Button/Button'
import { Spacer } from '../Spaces/Spaces'
import { Heading1 } from '../Typography/Headings'
import { Text } from '../Typography/Paragraph'
import { ModalBackground, Modal as ModalBox } from './Modal.styles'

interface ModalProps {
  title: string
  content?: string
  visible: boolean
  onClose: Function
}

function Modal({ title, content, visible, onClose }: ModalProps) {
  const [showModal, setShowModal] = useState(visible)

  useEffect(() => {
    setShowModal(visible)
  }, [visible])

  return (
    <>
      <ModalBackground visible={showModal}>
        <ModalBox>
          <Heading1>{title}</Heading1>
          <Text>{content}</Text>
          <Spacer size="big" />
          <Button
            label="Ok, tentar mais tarde"
            onClick={() => {
              setShowModal(false)
              onClose()
            }}
          />
        </ModalBox>
      </ModalBackground>
    </>
  )
}

export default Modal
