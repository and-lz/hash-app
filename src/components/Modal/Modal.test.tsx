import { fireEvent, render, RenderResult } from '@testing-library/react'
import React from 'react'
import { act } from 'react-dom/test-utils'
import Modal from './Modal'

describe('<Modal />', () => {
  let component: RenderResult
  let modalTitle: HTMLElement | null
  let modalContent: HTMLElement | null
  let modalButtonAction: Node | Window
  beforeEach(() => {
    component = render(
      <Modal
        title="Basic title"
        visible
        content="Basic content"
        action="OK, close"
        onClose={() => {}}
      />,
    )
    modalTitle = component.queryByTestId('modal-title')
    modalContent = component.queryByTestId('modal-content')
    modalButtonAction = component.getByText('OK, close')
  })
  test('Should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  test('Should render modal with the correct title, content, and button', () => {
    expect(modalTitle).toHaveTextContent('Basic title')
    expect(modalContent).toHaveTextContent('Basic content')
    expect(modalButtonAction).toBeInTheDocument()
  })
  test('Should not render modal when property visible was false', () => {
    component.rerender(
      <Modal title="Basic title" content="Basic content" action="OK, close" />,
    )
    expect(modalTitle).not.toBeInTheDocument()
    expect(modalContent).not.toBeInTheDocument()
  })
  test('Should hide modal after click on action button', () => {
    expect(modalTitle).toBeInTheDocument()
    expect(modalContent).toBeInTheDocument()

    act(() => {
      fireEvent.click(modalButtonAction)
    })

    expect(modalTitle).not.toBeInTheDocument()
    expect(modalContent).not.toBeInTheDocument()
  })
})
