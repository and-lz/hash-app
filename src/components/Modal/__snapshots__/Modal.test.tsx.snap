// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKckTs hxdzqV"
      >
        <div
          class="sc-iCfLBT bzdGJp"
        >
          <img
            class="sc-pVTma fpCxsv"
            src="images/header.jpg"
          />
          <div
            class="sc-furvIG fSdGQK"
          >
            <h1
              class="sc-dkPtyc cjJKLj"
              data-testid="modal-title"
            >
              Basic title
            </h1>
            <span
              class="sc-gsDJrp gvaAwY"
            />
            <p
              class="sc-eCImvq dQlTUo"
              data-testid="modal-content"
            >
              Basic content
            </p>
            <span
              class="sc-gsDJrp jEEgNF"
            />
            <button
              class="sc-bdvvaa bjJSUn"
            >
              OK, close
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKckTs hxdzqV"
    >
      <div
        class="sc-iCfLBT bzdGJp"
      >
        <img
          class="sc-pVTma fpCxsv"
          src="images/header.jpg"
        />
        <div
          class="sc-furvIG fSdGQK"
        >
          <h1
            class="sc-dkPtyc cjJKLj"
            data-testid="modal-title"
          >
            Basic title
          </h1>
          <span
            class="sc-gsDJrp gvaAwY"
          />
          <p
            class="sc-eCImvq dQlTUo"
            data-testid="modal-content"
          >
            Basic content
          </p>
          <span
            class="sc-gsDJrp jEEgNF"
          />
          <button
            class="sc-bdvvaa bjJSUn"
          >
            OK, close
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
